src/mesh/rbf.f90

```diff
@@ -0,0 +1,597 @@
+        ! Check if node is inside the box
+        x = Side%CurvedNode(QuadMapInv(0,0))%np%x
+        IF ((x(1).LT.xMin).OR.(x(1).GT.xMax).OR.(x(2).LT.yMin).OR.(x(2).GT.yMax)) EXIT
+        ! Increase count of boundary points if inside the box
+        nBP = nBP + 1
+        ! Mark this side node as done
+        Side%CurvedNode(QuadMapInv(0,0))%np%tmp = 1
+      END IF
+      IF (.NOT.(Side%CurvedNode(QuadMapInv(0,N))%np%tmp.EQ.1)) THEN
+        x = Side%CurvedNode(QuadMapInv(0,0))%np%x
+        IF ((x(1).LT.xMin).OR.(x(1).GT.xMax).OR.(x(2).LT.yMin).OR.(x(2).GT.yMax)) EXIT
+        nBP = nBP + 1
+        Side%CurvedNode(QuadMapInv(0,N))%np%tmp = 1
+      END IF
+      IF (.NOT.(Side%CurvedNode(QuadMapInv(N,0))%np%tmp.EQ.1)) THEN
+        x = Side%CurvedNode(QuadMapInv(0,0))%np%x
+        IF ((x(1).LT.xMin).OR.(x(1).GT.xMax).OR.(x(2).LT.yMin).OR.(x(2).GT.yMax)) EXIT
+        nBP = nBP + 1
+        Side%CurvedNode(QuadMapInv(N,0))%np%tmp = 1
+      END IF
+      IF (.NOT.(Side%CurvedNode(QuadMapInv(N,N))%np%tmp.EQ.1)) THEN
+        x = Side%CurvedNode(QuadMapInv(0,0))%np%x
+        IF ((x(1).LT.xMin).OR.(x(1).GT.xMax).OR.(x(2).LT.yMin).OR.(x(2).GT.yMax)) EXIT
+        nBP = nBP + 1
+        Side%CurvedNode(QuadMapInv(N,N))%np%tmp = 1
+      END IF
+    ELSE IF(Side%BC%BCType.EQ.1) THEN
+      ! Periodic BC, threat like a normal (non-BC) side
+      IF (.NOT.(Side%CurvedNode(QuadMapInv(0,0))%np%tmp.EQ.1)) THEN
+        x = Side%CurvedNode(QuadMapInv(0,0))%np%x
+        IF ((x(1).LT.xMin).OR.(x(1).GT.xMax).OR.(x(2).LT.yMin).OR.(x(2).GT.yMax)) EXIT
+        nBP = nBP + 1
+        Side%CurvedNode(QuadMapInv(0,0))%np%tmp = 1
+      END IF
+      IF (.NOT.(Side%CurvedNode(QuadMapInv(0,N))%np%tmp.EQ.1)) THEN
+        x = Side%CurvedNode(QuadMapInv(0,0))%np%x
+        IF ((x(1).LT.xMin).OR.(x(1).GT.xMax).OR.(x(2).LT.yMin).OR.(x(2).GT.yMax)) EXIT
+        nBP = nBP + 1
+        Side%CurvedNode(QuadMapInv(0,N))%np%tmp = 1
+      END IF
+      IF (.NOT.(Side%CurvedNode(QuadMapInv(N,0))%np%tmp.EQ.1)) THEN
+        x = Side%CurvedNode(QuadMapInv(0,0))%np%x
+        IF ((x(1).LT.xMin).OR.(x(1).GT.xMax).OR.(x(2).LT.yMin).OR.(x(2).GT.yMax)) EXIT
+        nBP = nBP + 1
+        Side%CurvedNode(QuadMapInv(N,0))%np%tmp = 1
+      END IF
+      IF (.NOT.(Side%CurvedNode(QuadMapInv(N,N))%np%tmp.EQ.1)) THEN
+        x = Side%CurvedNode(QuadMapInv(0,0))%np%x
+        IF ((x(1).LT.xMin).OR.(x(1).GT.xMax).OR.(x(2).LT.yMin).OR.(x(2).GT.yMax)) EXIT
+        nBP = nBP + 1
+        Side%CurvedNode(QuadMapInv(N,N))%np%tmp = 1
+      END IF
+    ELSE IF (Side%CurveIndex.LE.0) THEN
+      ! Boundaries, but uncurved - all nodes are fixed
+      x = Side%CurvedNode(QuadMapInv(0,0))%np%x
+      IF ((x(1).LT.xMin).OR.(x(1).GT.xMax).OR.(x(2).LT.yMin).OR.(x(2).GT.yMax)) EXIT
+      DO i=1,Side%nCurvedNodes
+        IF (Side%CurvedNode(i)%np%tmp.EQ.1) CYCLE
+        nBP = nBP + 1
+        Side%CurvedNode(i)%np%tmp = 1
+      END DO
```

Comment:
Why do we need to add the uncurved nodes to the `Side%CurvedNode` array?

Reply:
The `CurvedNode` array contains the high-order nodes. If the side is initially uncurved, i.e. linear, we add all of its nodes, since we want to curve it using the RBF volume curving.